src/Compute/Compute/Manual/VirtualMachineScaleSetRollingExtensionStartUpgradeNon.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Compute.Automation.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System.Threading.Tasks;
+
+
+namespace Microsoft.Azure.Commands.Compute.Automation
+{
+    [Cmdlet(VerbsLifecycle.Start, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VmssRollingExtensionUpgradeNon", DefaultParameterSetName = "DefaultParameter", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSOperationStatusResponse))]
+    public partial class VirtualMachineScaleSetRollingExtensionStartUpgradeNon : ComputeAutomationBaseCmdlet
+    {
+        private const string ByResourceIdParamSet = "ByResourceId",
+            ByInputObjectParamSet = "ByInputObject";
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceNameCompleter("Microsoft.Compute/virtualMachineScaleSets", "ResourceGroupName")]
```

Comment:
since this is for VMScaleSetName parameter, "ResourceGroupName" completer is not needed

Reply:
@grizzlytheodore , based on its usage [here](https://github.com/Azure/azure-powershell/blob/master/src/Compute/Compute/Generated/VirtualMachineScaleSetRollingUpgrade/VirtualMachineScaleSetRollingUpgradeStartOSUpgradeMethod.cs), I think this type of completer looks at the available VMSS resources and completes the name for those found in the given RGName.

Reply:
That functionality would fit here.